clients/src/main/java/org/apache/kafka/common/protocol/Errors.java

```diff
@@ -87,6 +89,13 @@
  * Do not add exceptions that occur only on the client or only on the server here.
  */
 public enum Errors {
+    UNKNOWN_RETRIABLE(-2, "The client received an unexpected error code when processing the response",
```

Comment:
Sure. I have renamed the variable `UNKNOWN` to `UNKNOWN_SERVER_ERROR` to distinguish it from `UNKNOWN_RETRIABLE`. I think it is reasonable because the corresponding exception is `UnknownServerException`. If we want to further differentiate the two, we also rename `UNKNOWN_RETRIABLE` to be `UNKNOWN_CLIENT_RETRIABLE` and rename `UnknownRetriableException` to `UnknownClientRetriableException`. What do you think?

And I added the following comment for `UNKNOWN_RETRIABLE`. Does that look OK?

```
// UNKNOWN_RETRIABLE can only occur on the client side when client's library version is lower than the server's version.
// This is used to allow client to retry if the error code in the response is new and not recognized.
```